[scarthgap] rpi-base.inc: vc4-kms-dsi-ili9881-7inch.dtbo

leon-anavi opened the pull request:
Add vc4-kms-dsi-ili9881-7inch.dtbo for the Raspberry Pi Touch Display 2 to RPI_KERNEL_DEVICETREE_OVERLAYS.

This work was sponsored by GOVCERT.LU.

<!--
Please make sure you've read and understood our contributing guidelines.

For additional information on the contribution guidelines:
https://wiki.yoctoproject.org/wiki/Contribution_Guidelines#General_Information

If this PR fixes an issue, make sure your description includes "fixes #xxxx".

If this PR connects to an issue, make sure your description includes "connected to #xxxx".

Please provide the following information:
-->

**- What I did**

Enabled support for Raspberry Pi Touch Display 2

**- How I did it**

Added `vc4-kms-dsi-ili9881-7inch.dtbo` to `RPI_KERNEL_DEVICETREE_OVERLAYS`.

To load the dtbo by default at the following line to your build setup, for example in `local.conf`:

```
RPI_EXTRA_CONFIG:append = " \
dtoverlay=vc4-kms-dsi-ili9881-7inch \
"
```

By default Raspberry Pi Touch Display 2 works in portrait mode. To rotate it in landscape mode for Weston (from `core-image-weston`), Add the following configuration to `weston.ini`:

```
[output]
name=DSI-2
mode=720x1280@60
transform=rotate-90 
```

Please note that in this example Raspberry Pi Touch Display 2 has been connected to physical connector `CAM/DISP 1` which was recognized and reported in the Weston log as `DSI-2`.

richard-tomlinson-red commented:
I have tried this change but I am getting nothing from the Touch Display 2 on my Raspberry Pi 4. I have tried both core-image-sato and core-image-weston. I've set MACHINE = "raspberrypi4-64". Is there any other configuration that's needed?
This is with the scarthgap branch on both poky and meta-raspberrypi.

The display works fine with Raspberry Pi OS.

leon-anavi commented:
> RPI_EXTRA_CONFIG:append = " \
> dtoverlay=vc4-kms-dsi-ili9881-7inch \
> "

Did you add the following line in your build setup?

RPI_EXTRA_CONFIG:append = " \
dtoverlay=vc4-kms-dsi-ili9881-7inch \
"

I have tested it successfully with Raspberry Pi 5 and it works fine on my side.

richard-tomlinson-red commented:
I added

`RPI_EXTRA_CONFIG:append = "dtoverlay=vc4-kms-dsi-ili9881-7inch"`

Is the slight change in format important?

leon-anavi commented:
> I added
> 
> `RPI_EXTRA_CONFIG:append = "dtoverlay=vc4-kms-dsi-ili9881-7inch"`
> 
> Is the slight change in format important?

Hm, yes, I think so because otherwise probably it won't be on a new line. Is ` In the `dtoverlay=vc4-kms-dsi-ili9881-7inch` on a separate line at `config.txt` from the boot partition of the generated image?

richard-tomlinson-red commented:
The end of config.txt:
`# Enable VC4 Graphics
dtoverlay=vc4-kms-v3d
dtoverlay=vc4-kms-dsi-ili9881-7inch
`
So definitely on a new line.

You are using Pi 5. Perhaps that is significant?

leon-anavi commented:
> End of config.txt:
> 
> ```bb
> # Enable VC4 Graphics
> dtoverlay=vc4-kms-v3d
> dtoverlay=vc4-kms-dsi-ili9881-7inch
> ```
> 
> Definitely on a new line.
> 
> You are using Pi 5. Perhaps that is significant?

Today I tested builds based on both master and scarthgap on Raspberry Pi 4 (64-bit) both worked fine. Weston starts. Here is DSI info from dmesg:
```
root@raspberrypi4-64:~# dmesg | grep dsi
[    0.045049] platform fe101000.cprman: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    0.047502] platform fe101000.cprman: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    0.047582] platform fe700000.dsi: Fixed dependency cycle(s) with /soc/dsi@7e700000/dsi_panel@0
[    0.047609] platform fe700000.dsi: Fixed dependency cycle(s) with /soc/cprman@7e101000
[    1.338428] vc4_dsi fe700000.dsi: Fixed dependency cycle(s) with /soc/dsi@7e700000/dsi_panel@0
[    1.338522] mipi-dsi fe700000.dsi.0: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    7.117270] ili9881c-dsi fe700000.dsi.0: supply power not found, using dummy regulator
[    7.343243] vc4-drm gpu: bound fe700000.dsi (ops vc4_dsi_ops)
```

Are you building for machine `raspberrypi4-64` (as in my tests) or `raspberrypi4` 32-bit version?

leon-anavi commented:
With this patch and the mentioned configurations the Raspberry Pi Touch Display 2 also works for `raspberrrypi4`. Here is a short log snippet:
```
root@raspberrypi4:~# cat /etc/os-release 
ID=poky
NAME="Poky (Yocto Project Reference Distro)"
VERSION="5.0.8 (scarthgap)"
VERSION_ID=5.0.8
VERSION_CODENAME="scarthgap"
PRETTY_NAME="Poky (Yocto Project Reference Distro) 5.0.8 (scarthgap)"
CPE_NAME="cpe:/o:openembedded:poky:5.0.8"
root@raspberrypi4:~# dmesg | grep dsi
[    0.043924] platform fe101000.cprman: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    0.046602] platform fe101000.cprman: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    0.046694] platform fe700000.dsi: Fixed dependency cycle(s) with /soc/dsi@7e700000/dsi_panel@0
[    0.046726] platform fe700000.dsi: Fixed dependency cycle(s) with /soc/cprman@7e101000
[    1.464840] vc4_dsi fe700000.dsi: Fixed dependency cycle(s) with /soc/dsi@7e700000/dsi_panel@0
[    1.464940] mipi-dsi fe700000.dsi.0: Fixed dependency cycle(s) with /soc/dsi@7e700000
[    8.123792] ili9881c-dsi fe700000.dsi.0: supply power not found, using dummy regulator
[    8.354882] vc4-drm gpu: bound fe700000.dsi (ops vc4_dsi_ops)
```

leon-anavi commented:
> > Hi @kraj,
> > All tested have passed but 3 haven't been executed because:
> > > The job has exceeded the maximum execution time while awaiting a runner for 24h0m0s
> > 
> > 
> > Could you please have a look as we still need this patch in Scarthgap?
> > Thanks, Leon
> 
> Can you rebase and push again ?

Yes, I've just force-pushed the commit.

kraj commented:
Can this be closed ? now that #1477 is merged

leon-anavi commented:
> Can this be closed ? now that #1477 is merged

No, we need this in Scarthgap to support the Raspberry Pi Touch Display 2.

leon-anavi commented:
@kraj all checks have passed. Could you please consider merging this pull request? It is a backport to Scarthgap from what we have already merged in branch master.
